alerting/src/main/kotlin/org/opensearch/alerting/triggercondition/parsers/TriggerExpressionParser.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.triggercondition.parsers
+
+import org.opensearch.alerting.triggercondition.resolvers.TriggerExpressionRPNResolver
+import org.opensearch.alerting.triggercondition.tokens.TriggerExpressionOperator
+
+/**
+ * The postfix (Reverse Polish Notation) parser.
+ * Uses the Shunting-yard algorithm to parse a mathematical expression
+ * @param triggerExpression String containing the trigger expression for the monitor
+ */
+class TriggerExpressionParser(
```

Comment:
should the class name be changed to `TriggerRPNExpressionParser`? 
`parse()` returns `TriggerExpressionRPNResolver`
tomorrow if someone wants to have another implementation of interface `ExpressionParser`

Reply:
Since currently there is only one concrete implementation of the Trigger Expression Parser have kept the name as is for clarity. However, the base implementation which holds the RPN logic is named as the ``TriggerExpressionRPNBaseParser``. 
Later addition of new base implementations are possible using the new strategy, while the concrete implementation could still expose them via different method interfaces.